core/templates/dev/head/domain/statistics/LearnerAnswerDetailsObjectFactory.ts

```diff
@@ -17,38 +17,57 @@
  * domain objects.
  */
 
-angular.module('oppia').factory('LearnerAnswerDetailsObjectFactory', [
-  function() {
-    var LearnerAnswerDetails = function(
-        expId, stateName, interactionId, customizationArgs,
-        learnerAnswerInfoData) {
-      this.expId = expId;
-      this.stateName = stateName;
-      this.interactionId = interactionId;
-      this.customizationArgs = customizationArgs;
-      this.learnerAnswerInfoData = learnerAnswerInfoData;
-    };
-
-    LearnerAnswerDetails.prototype.getExpId = function() {
-      return this.expId;
-    };
-
-    LearnerAnswerDetails.prototype.getStateName = function() {
-      return this.stateName;
-    };
-
-    LearnerAnswerDetails.prototype.getLearnerAnswerInfoData = function() {
-      return this.learnerAnswerInfoData;
-    };
-
-    /* eslint-disable dot-notation */
-    LearnerAnswerDetails['createDefaultLearnerAnswerDetails'] = function(expId,
-        stateName, interactionId, customizationArgs, learnerAnswerInfoData) {
-    /* eslint-enable dot-notation */
-      return new LearnerAnswerDetails(expId, stateName, interactionId,
-        customizationArgs, learnerAnswerInfoData);
-    };
-
-    return LearnerAnswerDetails;
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+
+import { LearnerAnswerInfo } from
+  'domain/statistics/LearnerAnswerInfoObjectFactory';
+
+export class LearnerAnswerDetails {
+  expId: string;
+  stateName: string;
+  interactionId: string;
+  customizationArgs: any;
+  learnerAnswerInfoData: LearnerAnswerInfo[];
+
+  constructor(
+      expId: string, stateName: string, interactionId: string,
+      customizationArgs: any,
+      learnerAnswerInfoData: LearnerAnswerInfo[]) {
+    this.expId = expId;
+    this.stateName = stateName;
+    this.interactionId = interactionId;
+    this.customizationArgs = customizationArgs;
+    this.learnerAnswerInfoData = learnerAnswerInfoData;
+  }
+
+  getExpId(): string {
+    return this.expId;
+  }
+
+  getStateName(): string {
+    return this.stateName;
+  }
+
+  getLearnerAnswerInfoData(): LearnerAnswerInfo[] {
+    return this.learnerAnswerInfoData;
   }
-]);
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class LearnerAnswerDetailsObjectFactory {
+  createDefaultLearnerAnswerDetails(
```

Comment:
Please make this static.

Reply:
Could you please let me know the reason for this? 

If I make this static -- it won't be available on the instance of class, rather than it will be available on the class and $provide.value() does not inject classes rather it injects class instances. I was getting whole lot of errors when I used class in $provide.value.

Reply:
The reason is mostly code health related -- there should be no need to create an instance in order to call the method. If there's truly no way to call it, however, I'm fine with keeping it as an instance method.

Reply:
I did tried it calling through the class name but its not possible -- I'm unable to inject a class though I can inject class instance. 

Apologies for the delay -- I was out for my convocation the last week.

Reply:
Hey @brianrodri, ping.